roles/ee_builder/templates/execution_environment.yml.j2

```diff
@@ -26,37 +26,38 @@ build_arg_defaults:
 ansible_config: '{{ ansible_config }}'
 {% endif %}
 
-{% if (ee_bindep is defined and ee_bindep|length ) or
-        (ee_python is defined and ee_python|length ) or
-        (ee_collections is defined and ee_collections|length) %}
+{% if (__execution_environment_defintion.bindep is defined and __execution_environment_defintion.bindep|length ) or
+        (__execution_environment_defintion.python is defined and __execution_environment_defintion.python|length ) or
+        (__execution_environment_defintion.collections is defined and __execution_environment_defintion.collections|length) or
+        (__execution_environment_defintion.roles is defined and __execution_environment_defintion.roles|length) %}
 dependencies:
-{% if ee_bindep is defined %}
-{%      if ee_bindep|length %}
-  system: {{ bindep_file }}
-{%      endif %}
+{% if (__execution_environment_defintion.bindep is defined and __execution_environment_defintion.bindep|length )%}
+  system: bindep
```

Comment:
If you could put these back to the variable that would be better to allow overriding if preferred.